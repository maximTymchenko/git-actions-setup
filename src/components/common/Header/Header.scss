@import '../../../styles/vars';

.header {
    background-color: $background__primary;
    margin: 0 auto;
    width: 100%;
    border-bottom: 2px solid $secondary;
    position: fixed;
    top: 0px;
    z-index: 999;

    display: flex;
    justify-content: center;

    @media screen and (max-width: $tablet) {
        justify-content: flex-start;
    }

    &__wrapper {
        display: flex;
        @include main__padding;
        width: 100%;
        // max-width: $header__width;
        align-items: center;

        @media screen and (max-width: $tablet) {
            width: 820px;
            justify-content: space-between;
        }

        @media screen and (max-width: $small__tablet) {
            width: 690px;
        }

        @media screen and (max-width: $large__mobile) {
            width: 630px;
        }

        @media screen and (max-width: $mobile) {
            width: 560px;
        }
    }
    &__logo {
        height: 100%;

        &--link {
            height: 100%;
            max-height: 63px;
        }

        @media screen and (max-width: $mobile) {
            padding-left: 0;
            margin: 0;
        }
    }

    &__button {
        @media screen and (max-width: $mobile) {
            width: 120px;
        }
    }

    &__nav {
        display: flex;
        align-items: center;
        justify-content: center;
        flex: 1;

        @media screen and (max-width: $tablet) {
            flex: 0;
        }

        &-wrapper {
            display: flex;
            align-items: center;
            flex: 1;

            &--end {
                justify-content: flex-end;
            }
        }

        &-text {
            cursor: pointer;
            color: $secondary;
            font-family: 'OktaNeue-Black';
            font-size: 15px;
            font-weight: bold;
            text-align: center;
            border-bottom: 4px solid transparent;
            transition: 0.3s linear all;

            @media screen and (max-width: $tablet) {
                display: none;
            }

            &:hover {
                color: $primary__light;
            }

            &--active {
                position: relative;
                top: 2px;
            }

            &--secondary {
                color: $text__body;
                position: relative;

                @media screen and (max-width: $tablet) {
                    display: none;
                }
            }

            &--unvisible {
                opacity: 0;

                @media screen and (max-width: $tablet) {
                    display: none;
                }
            }

            &--hidden {
                display: none;
            }
        }
        &-menu {
            cursor: pointer;
            outline: none;
            width: 36px;
            height: 36px;
            margin: 0 20px 0px 20px;
            border: solid 2px $secondary;
            background-color: $background__primary;
            display: flex;
            align-items: center;
            justify-content: center;
            flex-direction: column;
            box-shadow: 3px 3px 1px 1px rgba(57, 156, 203, 0.4);
            @include transition;

            @media screen and (max-width: $mobile) {
                margin: 0;
            }

            &:hover {
                background-color: $background__secondary;
            }
            &:active {
                @include button__press;
                box-shadow: 3px 3px 1px 1px rgba(57, 156, 203, 0);
                border-radius: none;
                background-color: $text__body--light;
            }
        }
        &-bars {
            margin: 1px;
            width: 14px;
            height: 3px;
            background-color: $secondary;
        }
        &-times {
            font-size: 20px;
        }
    }

    &__options {
        display: flex;
        // margin-right: 100px;

        @media screen and (max-width: $tablet) {
            margin-right: 0px;
        }
    }

    &__options-list {
        cursor: pointer;
        outline: none;
        border: none;
        background-color: transparent;
        font-family: 'OktaNeue-SemiBold';
        font-weight: 600;
        letter-spacing: 0.3px;
        color: $text__body;
        font-size: 15px;
        letter-spacing: 0.3px;

        @media screen and (max-width: $tablet) {
            display: none;
        }
    }
}

//! CHEVRON

.chevron__icon {
    margin: 0 5px;
    font-size: 20px;
    color: $text__body;

    @media screen and (max-width: $tablet) {
        display: none;
    }
}
