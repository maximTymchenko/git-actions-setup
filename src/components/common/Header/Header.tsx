import React, { useState, FC } from 'react';
import { Link } from 'gatsby';
import { FaTimes } from 'react-icons/fa';
import { FiChevronLeft } from 'react-icons/fi';
import './Header.scss';
import Button from '../Button';
import logo from '../../../assets/img/logo.svg';
import arrowRight from '../../../assets/img/arr-right.svg';
import classNames from 'classnames';
import PrimaryNav from '../PrimaryNav';
import Nav from '../Nav';

type THeaderArrays = {
    activeLink: null;
    links: {
        id: number;
        title: string;
        page: string;
    }[];
};

type THeaderArraysOfObjects = {
    activeLink: null;
    links: {
        id: number;
        array: THeaderArrays;
    }[];
};

const Header: FC = (props: any) => {
    const [buttonIcon, setButtonIcon] = useState<boolean>(true);
    const [buttonIconTimes, setButtonIconTimes] = useState<boolean>(false);
    const [showMainNav, setShowMainNav] = useState<boolean>(false);

    const changeNavVisible = (): void => {
        setShowMainNav(!showMainNav);
    };

    const changeButtonIcon = (): void => {
        setButtonIcon(!buttonIcon);
        setButtonIconTimes(!buttonIconTimes);
        changeNavVisible();
    };

    const language = {
        activeLink: null,
        links: [
            { id: 1, title: 'EN' },
            { id: 2, title: 'UA' },
        ],
    };

    const [navLinks, setNavLinks] = useState<THeaderArrays>({
        activeLink: null,
        links: [
            { id: 1, title: 'Home' },
            { id: 2, title: 'How We Work' },
            { id: 3, title: 'What We Do' },
            { id: 4, title: 'About Us' },
            { id: 5, title: 'Contact Blackbird' },
            { id: 6, title: 'Careers' },
            { id: 7, title: 'Blog' },
        ],
    });

    const menuLinksHome: THeaderArrays = {
        activeLink: null,
        links: [{ id: 1, title: 'Home', page: '/' }],
    };

    const menuLinksHowWeWork: THeaderArrays = {
        activeLink: null,
        links: [
            { id: 1, title: 'Engagement Models', page: '/how-we-work/working-together/' },
            { id: 2, title: 'Your IP Protection', page: '/how-we-work/intellectual-property/' },
        ],
    };

    const menuLinksWhatWeDo: THeaderArrays = {
        activeLink: null,
        links: [
            { id: 1, title: 'All', page: '/what-we-do/' },
            { id: 2, title: 'Web development', page: '/what-we-do/web-development/' },
            { id: 3, title: 'Cloud services', page: '/what-we-do/cloud-services/' },
            { id: 4, title: 'Team extension', page: '/what-we-do/team-extension/' },
            { id: 5, title: 'AR & VR', page: '/what-we-do/ar-&-vr/' },
            { id: 6, title: 'Retail & E-commerce', page: '/what-we-do/retail-and-e-commerce/' },
            { id: 7, title: 'Mobile development', page: '/what-we-do/mobile-development/' },
        ],
    };

    const menuLinksAboutUs: THeaderArrays = {
        activeLink: null,
        links: [
            { id: 1, title: 'Retail & E-commerce' },
            { id: 2, title: 'Construction' },
            { id: 3, title: 'Finance' },
            { id: 4, title: 'Media' },
        ],
    };

    const menuLinksContact: THeaderArrays = {
        activeLink: null,
        links: [
            { id: 1, title: 'SaaS product' },
            { id: 2, title: 'Chatbots' },
            { id: 3, title: 'AR & VR' },
            { id: 4, title: 'Blockchain' },
            { id: 5, title: 'Digital enterprise ' },
            { id: 6, title: 'Drone' },
            { id: 7, title: 'R&D' },
        ],
    };

    const menuLinksCareers: THeaderArrays = {
        activeLink: null,
        links: [
            { id: 1, title: 'On-demand team extension' },
            { id: 2, title: 'Fully managed IT services' },
        ],
    };

    const menuLinksBlog: THeaderArrays = {
        activeLink: null,
        links: [
            { id: 1, title: 'Latest', page: '/blog/' },
            { id: 2, title: 'Editorial', page: '/blog/editorial' },
            { id: 3, title: 'Whitepapers', page: '/blog/whitepapers' },
            { id: 4, title: 'Videos', page: '/blog/videos' },
            { id: 5, title: 'Resources', page: '/blog/resources' },
            {
                id: 6,
                title: 'Press center',
                page: '/blog/press-center',
            },
            ,
        ],
    };

    //! show subnav with button click on main nav

    const [navLinksArray, setNavLinksArray] = useState<THeaderArraysOfObjects>({
        activeLink: null,
        links: [
            { id: 0, array: menuLinksHome },
            { id: 1, array: menuLinksHowWeWork },
            { id: 2, array: menuLinksWhatWeDo },
            { id: 3, array: menuLinksAboutUs },
            { id: 4, array: menuLinksContact },
            { id: 5, array: menuLinksCareers },
            { id: 6, array: menuLinksBlog },
        ],
    });

    return (
        <header data-testid="header" className="header">
            <div className="header__wrapper">
                <Link to="/" className="header__logo--link">
                    <img className="header__logo" src={logo} alt="BlackBird" />
                </Link>
                <div className="header__nav">
                    <div className="header__nav-wrapper header__nav-wrapper--end">
                        <Link
                            className="link"
                            // to="/working-together/"
                            activeClassName="active__border active__border--style"
                        >
                            <span
                                className={classNames('header__nav-text', {
                                    'active__border--style': props.path,
                                })}
                            >
                                {props.path}
                            </span>
                        </Link>

                        {props.path && <FiChevronLeft className="chevron__icon" />}
                        <Link
                            className="link"
                            to="/how-we-work/"
                            activeClassName="active__border active__border--style"
                        >
                            <span
                                className={classNames('header__nav-text', {
                                    'header__nav-text--secondary': props.path,
                                })}
                            >
                                How We Work
                            </span>
                        </Link>
                    </div>

                    <div className="header__nav-wrapper">
                        <button onClick={() => changeButtonIcon()} className="header__nav-menu">
                            {buttonIcon && (
                                <>
                                    <div className="header__nav-bars" />
                                    <div className="header__nav-bars" />
                                    <div className="header__nav-bars" />
                                </>
                            )}
                            {buttonIconTimes && <FaTimes className="header__nav-times" />}
                        </button>
                        {buttonIconTimes && (
                            <div className="primary-nav__section">
                                <div className="primary-nav">
                                    <ul className="primary-nav__wrapper">
                                        <PrimaryNav navTitle="Home" navArray={menuLinksHome} />
                                        <PrimaryNav
                                            navTitle="How We Work"
                                            navArray={menuLinksHowWeWork}
                                        />
                                        <PrimaryNav
                                            navTitle="What We Do"
                                            navArray={menuLinksWhatWeDo}
                                        />
                                        <PrimaryNav
                                            navTitle="About Us"
                                            navArray={menuLinksAboutUs}
                                        />
                                        <PrimaryNav
                                            navTitle="Contact Blackbird"
                                            navArray={menuLinksContact}
                                        />
                                        <PrimaryNav
                                            navTitle="Careers"
                                            navArray={menuLinksCareers}
                                        />
                                        <PrimaryNav navTitle="Blog" navArray={menuLinksBlog} />
                                    </ul>
                                    <div className="primary-nav__language">
                                        <Nav type="" links={language} />
                                    </div>
                                </div>
                            </div>
                        )}
                        <Link
                            className="link"
                            partiallyActive={true}
                            activeClassName="active__border active__border--style"
                            to="/what-we-do/"
                        >
                            <span className="header__nav-text">What We Do</span>
                        </Link>
                    </div>
                </div>
                <div className="header__options">
                    <div className="header__button">
                        <Link className="link" to="/contact-us">
                            <Button
                                icon={arrowRight}
                                type="secondary"
                                className="button"
                                desc="Contact us"
                                mobile="mobile"
                            >
                                Contact us
                            </Button>
                        </Link>
                    </div>
                    <select className="header__options-list" id="languages">
                        <option defaultValue="en">EN</option>
                        <option value="ukr">UKR</option>
                        <option value="rus">RUS</option>
                    </select>
                </div>
            </div>
        </header>
    );
};

export default Header;
