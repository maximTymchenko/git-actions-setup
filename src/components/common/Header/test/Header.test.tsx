import React from 'react';
import Header from '../Header';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<Header path="test path"></Header>);
    expect(getByTestId('header')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<Header path="test path"></Header>).toJSON();
    expect(tree).toMatchSnapshot();
});
