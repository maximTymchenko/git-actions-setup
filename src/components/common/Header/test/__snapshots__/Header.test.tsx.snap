// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<header
  className="header"
  data-testid="header"
>
  <div
    className="header__wrapper"
  >
    <a
      aria-current="page"
      className="header__logo--link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      <img
        alt="BlackBird"
        className="header__logo"
        src="test-file-stub"
      />
    </a>
    <div
      className="header__nav"
    >
      <div
        className="header__nav-wrapper header__nav-wrapper--end"
      >
        <a
          className="link"
        >
          <span
            className="header__nav-text active__border--style"
          >
            test path
          </span>
        </a>
        <svg
          className="chevron__icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="15 18 9 12 15 6"
          />
        </svg>
        <a
          className="link"
          href="/how-we-work/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            className="header__nav-text header__nav-text--secondary"
          >
            How We Work
          </span>
        </a>
      </div>
      <div
        className="header__nav-wrapper"
      >
        <button
          className="header__nav-menu"
          onClick={[Function]}
        >
          <div
            className="header__nav-bars"
          />
          <div
            className="header__nav-bars"
          />
          <div
            className="header__nav-bars"
          />
        </button>
        <a
          className="link"
          href="/what-we-do/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            className="header__nav-text"
          >
            What We Do
          </span>
        </a>
      </div>
    </div>
    <div
      className="header__options"
    >
      <div
        className="header__button"
      >
        <a
          className="link"
          href="/contact-us"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <button
            className="button button--secondary button--mobile button--undefined"
            data-testid="button"
          >
            Contact us
            <img
              alt="Contact us"
              className="button__icon button__icon--undefined"
              data-testid="button__icon"
              src="test-file-stub"
            />
          </button>
        </a>
      </div>
      <select
        className="header__options-list"
        id="languages"
      >
        <option
          defaultValue="en"
        >
          EN
        </option>
        <option
          value="ukr"
        >
          UKR
        </option>
        <option
          value="rus"
        >
          RUS
        </option>
      </select>
    </div>
  </div>
</header>
`;
